src/garage/envs/gym_env.py

```diff
@@ -0,0 +1,409 @@
+
+    def step(self, action):
+        """Call step on wrapped env.
+
+        Args:
+            action (np.ndarray): An action provided by the agent.
+
+        Returns:
+            EnvStep: The time step resulting from the action.
+
+        Raises:
+            RuntimeError: if `step()` is called after the environment has been
+                constructed and `reset()` has not been called.
+
+        """
+        if self._last_observation is None:
+            raise RuntimeError('reset() must be called before step()!')
+
+        observation, reward, done, info = self._env.step(action)
+
+        if self._visualize:
+            self._env.render(mode='human')
+
+        last_obs = self._last_observation
+        # Type conversion
+        if not isinstance(reward, float):
+            reward = float(reward)
+
+        self._last_observation = observation
+        self._step_cnt += 1
+
+        step_type = None
+        if done:
+            step_type = StepType.TERMINAL
+        elif self._step_cnt == 1:
+            step_type = StepType.FIRST
+        else:
+            step_type = StepType.MID
+
+        # gym envs that are wrapped in TimeLimit wrapper modify
+        # the done/termination signal to be true whenever a time
+        # limit expiration occurs. The following statement sets
+        # the done signal to be True only if caused by an
+        # environment termination, and not a time limit
+        # termination. The time limit termination signal
+        # will be saved inside env_infos as
+        # 'GymEnv.TimeLimitTerminated'
+        if (self._step_cnt >= self._spec.max_episode_length
+                or 'TimeLimit.truncated' in info):
+            info['GymEnv.TimeLimitTerminated'] = True
+            step_type = StepType.TIMEOUT
+        else:
+            info['TimeLimit.truncated'] = False
+            info['GymEnv.TimeLimitTerminated'] = False
+
+        return EnvStep(env_spec=self.spec,
+                       observation=last_obs,
+                       action=action,
+                       reward=reward,
+                       next_observation=observation,
+                       env_info=info,
+                       step_type=step_type)
+
+    def render(self, mode):
+        """Renders the environment.
+
+        Args:
+            mode (str): the mode to render with. The string must be present in
+                `self.render_modes`.
+
+        Returns:
+            object: the return value for render, depending on each
+                environment's implementation.
+        """
+        self._validate_render_mode(mode)
+        return self._env.render(mode)
+
+    def visualize(self):
+        """Creates a visualization of the environment."""
+        self._env.render(mode='human')
+        self._visualize = True
+
+    def close(self):
+        """Close the wrapped env."""
+        self._close_viewer_window()
+        self._env.close()
+
+    def compute_reward(self, achieved_goal, desired_goal, info):
```

Comment:
This is a different subclass of gym.Env called a GoalEnv -- it should probably get it's own wrapper. Maybe you can do it the same way you supported BulletEnv?